okhttp/src/main/java/okhttp3/internal/Util.kt

```diff
@@ -0,0 +1,642 @@
+/*
+ * Copyright (C) 2012 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal
+
+import okhttp3.EventListener
+import okhttp3.Headers
+import okhttp3.HttpUrl
+import okhttp3.RequestBody
+import okhttp3.ResponseBody
+import okhttp3.internal.http2.Header
+import okhttp3.internal.platform.Platform
+import okio.Buffer
+import okio.BufferedSource
+import okio.ByteString.Companion.decodeHex
+import okio.Options
+import okio.Source
+import java.io.Closeable
+import java.io.IOException
+import java.io.InterruptedIOException
+import java.net.IDN
+import java.net.InetAddress
+import java.net.ServerSocket
+import java.net.Socket
+import java.nio.charset.Charset
+import java.nio.charset.StandardCharsets.UTF_16BE
+import java.nio.charset.StandardCharsets.UTF_16LE
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.ArrayList
+import java.util.Arrays
+import java.util.Arrays.asList
+import java.util.Collections
+import java.util.Comparator
+import java.util.LinkedHashMap
+import java.util.Locale
+import java.util.TimeZone
+import java.util.concurrent.ThreadFactory
+import java.util.concurrent.TimeUnit
+import javax.net.ssl.X509TrustManager
+
+/** Junk drawer of utility methods.  */
+object Util {
```

Comment:
drop the enclosing object and make these top-level? `@file:JvmName("Util")`

Reply:
Yep, but I will do in follow up as it changes a lot of other files.  Kotlin classes won't use Util.x